File: Ruby/k03.rb
class K03
  def execute(num)
    case num
    when 1
      question1
    when 2
      question2
    when 3
      question3
    when 4
      question4
    else
      return
    end
  end

  def question1
    begin
      print '年齢を入力してください。＞'
      age = gets.chomp.to_i
      puts '未成年なので購入できません。' if age < 20
    rescue TypeError => e
      puts e.message
    end
  end

  def question2
    begin
      print '身長を入力してください。＞'
      height = gets.chomp.to_f
      height /= 100.0
      print '体重を入力してください。＞'
      weight = gets.chomp.to_f
      standard = height * height * 22.0
      puts "あなたの標準体重は#{standard}です。"

      if weight > standard && (weight - standard) / standard * 100 > 14
        puts '太り気味です。'
      elsif weight < standard && (weight - standard) / standard * 100 < -14
        puts '痩せ気味です。'
      else
        puts '普通ですね。'
      end
    rescue TypeError => e
      puts e.message
    rescue ZeroDivisionError => e
      puts e.message
    end
  end

  def question3
    begin
      number = rand(0..99)
      puts '０から９９の範囲の数値が決定されました。'
      print '決められた数値を予想し、この数値よりも大きな値を入力してください＞'
      guess = gets.chomp.to_i
      puts "決められた数値は#{number}です。"
      puts guess > number ? '正解です。' : '不正解です。'
    rescue TypeError => e
      puts e.message
    end
  end

  def question4
    begin
      number = rand(0..99)
      puts '０から９９の範囲の数値が決定されました。'
      print '決められた数値を予想し、この数値よりも大きな値を入力してください＞'
      guess = gets.chomp.to_i
      puts "決められた数値は#{number}です。"

      if guess > 100 || guess < 0
        puts '反則です！'
      elsif guess > number && guess - number <= 10
        puts '大正解！'
      elsif guess < number && guess - number >= -10
        puts '惜しい！'
      elsif guess == number
        puts 'お見事！'
      else
        puts guess > number ? '正解です。' : '不正解です。'
      end
    rescue TypeError => e
      puts e.message
    end
  end
end

File: Cpp/K03.cpp
#include <ctime>
#include <iostream>
#include <random>
#include <stdexcept>

#include "K03.h"


K03::K03() = default;


K03::~K03() = default;

void K03::question1() {
    try {
        std::cout << "年齢を入力してください。＞";
        auto age = 0;
        std::cin >> age;

        if (age < 20) {
            std::cout << "未成年なので購入できません。\n";
        }
    }
    catch (const std::exception &ex) {
        std::cout << ex.what();
    }
}

void K03::question2() {
    try {
        std::cout << "身長を入力してください。＞";
        auto height = 0.0f;
        std::cin >> height;
        height /= 100.0f;
        std::cout << "体重を入力してください。＞";
        auto weight = 0.0f;
        std::cin >> weight;

        const auto standard = height * height * 22.0f;
        std::cout << "あなたの標準体重は" << standard << "です。\n";

        if (weight > standard && (weight - standard) / standard * 100.0f > 14.0f) {
            std::cout << "太り気味です。\n";
        } else if (weight < standard && (weight - standard) / standard * 100.0f < -14.0f) {
            std::cout << "痩せ気味です。\n";
        } else {
            std::cout << "普通ですね。\n";
        }
    }
    catch (const std::exception &ex) {
        std::cout << ex.what();
    }

}

void K03::question3() {
    try {
        std::mt19937 generator(static_cast<std::mt19937::result_type>(std::time(nullptr)));
        std::uniform_int_distribution<> randNumGen(0, 99);

        const auto randNum = randNumGen(generator);

        std::cout << "０から９９の範囲の数値が決定されました。\n";
        std::cout << "決められた数値を予想し、この数値よりも大きな値を入力してください＞";
        auto guess = 0;
        std::cin >> guess;

        std::cout << "決められた数値は" << randNum << "です。\n";
        std::cout << ((guess > randNum) ? "正解です。\n" : "不正解です。\n");
    }
    catch (const std::exception &ex) {
        std::cout << ex.what();
    }
}

void K03::question4() {
    try {
        std::mt19937 generator(static_cast<std::mt19937::result_type>(std::time(nullptr)));
        std::uniform_int_distribution<> randNumGen(0, 99);

        const auto randNum = randNumGen(generator);

        std::cout << "０から９９の範囲の数値が決定されました。\n";
        std::cout << "決められた数値を予想し、この数値よりも大きな値を入力してください＞";
        auto guess = 0;
        std::cin >> guess;

        std::cout << "決められた数値は" << randNum << "です。\n";

        if (guess < 0 || guess > 100) {
            std::cout << "反則です！";
        } else if (guess > randNum && guess - randNum <= 10) {
            std::cout << "大正解です！";
        } else if (guess < randNum && guess - randNum >= -10) {
            std::cout << "惜しい！";
        } else if (guess == randNum) {
            std::cout << "お見事！";
        } else {
            std::cout << ((guess > randNum) ? "正解です。\n" : "不正解です。\n");
        }
    }
    catch (const std::exception &ex) {
        std::cout << ex.what();
    }
}
